WIP: git workflows

Camwyn opened the pull request:
### 🗒️ Description

In a [previous PR](https://github.com/the-events-calendar/the-events-calendar/pull/5222) attempting to split up the views_integration tests, I made some changes to the workflow process.

I realized those should get pulled out into their own changeset. This is that and a bit more.

After converting what we have to leverage reusable parts, I then refactored the release tasks into reusable parts as well, allowing me to create one task orchestrating them all.

The next piece was to begin taking manual tasks that are currently not handled in our actions and adding them, thus a new reusable step equivalent to `tut analyze` - runs after the changelog is created and adds the changelog entries before processing. As a standalone, I've preserved the varied output formats, so they can still be  used to generate lists, etc for use elsewhere.


PS: Once this is working, my intention is to move as much of it as possible to`the-events-calendar/actions/.github/actions/`

Camwyn committed:
MT cleanup

Camwyn committed:
Transition from `master` to `main` base branch.

Camwyn committed:
Remove ${CARET} from docblocks.

These are from someone's malformed docblock template - they need to go.

Requires: https://github.com/the-events-calendar/tribe-common/pull/2637

Camwyn committed:
Leverage reusable pieces and templates for existing workflows.

Camwyn committed:
Refactor release process to use reusable workflows.

Camwyn committed:
Add a step equivalent to `tut analyze`

Camwyn committed:
Alter the analyze-changes workflow to run before process-changelog in the orchestrator, defaulting to changelog format.

Still allow the output format to be specified in the standalone workflow.

Camwyn committed:
Fix the architecture to stay DRY.

lucatume reviewed (changes requested):
It's a bit difficult to parse this after all the work in #5222.

This seems a more gradual approach that would be merged before #5222, correct?

Camwyn committed:
Remove extra comments.

Camwyn commented:
> It's a bit difficult to parse this after all the work in #5222.
> 
> This seems a more gradual approach that would be merged before #5222, correct?

Correct - I realized this piece was more digestible and more gradual.

I've closed #5222 as (from our conversations) I think the splitting is not the best path forward.

dpanta94 commented on .github/workflows/main-release.yml:
```diff
@@ -0,0 +1,187 @@
+        description: "Target branch for the workflow"
+        required: false
+        type: string
+        default: "main"
```
Probably the default should be master until we switch over

dpanta94 reviewed (changes requested):
In general i like what is being done here! Left some comments though

Camwyn committed:
Address some review comments.

Camwyn committed:
update common hash

Camwyn committed:
update common hash

Camwyn committed:
Streamline the analyze-changes workflow.

Camwyn committed:
MOve most of this out into our actions repo

Camwyn committed:
Update common hash

Camwyn committed:
Correct process-changelog.sh to use the new version.

Camwyn committed:
feat: Add tec-actions configuration and improve changelog processing

- Add skip-flags configuration to package.json under tec-actions
- Configure views-full-suite and default skip flags for test optimization
- Update process-changelog.sh to parse changelog_url from package.json
- Add package.json parsing logic to read tec-actions.changelog_url
- Support fallback to parameter or default URL when package.json not configured
- Change config key from 'tec' to 'tec-actions' for better namespacing

Camwyn committed:
Alter the skip process in tests-php.yml to use the new skip-flags-parser.yml.
Allow it to use a global and suite-specific skip flag.
